src/org/omegat/gui/glossary/GlossaryManager.java

```diff
@@ -155,7 +155,8 @@ public void fileChanged(File file) {
         }
         if (file.exists()) {
             try {
-                List<GlossaryEntry> entries = loadGlossaryFile(file);
+                boolean isPriority = priorityGlossary != null && priorityGlossary.equals(file);
```

Comment:
I feel like I'm missing something here: what's the benefit of doing this check earlier?

Reply:
When OmegaT open new team project, and downloading the project, OmegaT does

1. run `RealProject#loadProject`  it comes a code block `RealProject` line 379

```java
               // This MUST happen after calling loadTranslations()
                if (remoteRepositoryProvider != null && isOnlineMode) {
                    Core.getMainWindow().showStatusMessageRB("TEAM_REBASE_AND_COMMIT");
                    rebaseAndCommitProject(true);
                }
```

Then `RealProject#rebaseAndCommitProject`  line 1020
There is a code block
```java

        if (processGlossary) {
            final String glossaryPath = config.getWritableGlossaryFile().getUnderRoot();
            final File glossaryFile = config.getWritableGlossaryFile().getAsFile();
            if (glossaryPath != null && remoteRepositoryProvider.isUnderMapping(glossaryPath)) {
                final List<GlossaryEntry> glossaryEntries;
```
where syncing remote glossary to local.

in line 1084
```java
                RebaseAndCommit.rebaseAndCommit(glossaryPrepared, remoteRepositoryProvider,
                        config.getProjectRootDir(), glossaryPath, new RebaseAndCommit.IRebase() {
```
calls `RebaseAndCommit.rebaseAndCommit` 

I proposes a change in the PR of a rebase and commit process. 
```java
                            @Override
                            public void reload(final File file) {
+                               logger.atDebug().setMessage("Reloading glossary file {0}").addArgument(file).log();
+                               notifyGlossaryManagerFileChanged(file);
                            }

    protected void notifyGlossaryManagerFileChanged(File file) {
        GlossaryManager gm = Core.getGlossaryManager();
        if (gm != null) {
            gm.fileChanged(file); // XXX
        }
    }


```
to call `GlossaryManager#fileChanged(file)`.
It is a timing when `GlossaryManager.start` has not started yet. It is a reason why we need to check whether `priorityGlossary` is initialized.

```java
    public void start() {
        File dir = new File(Core.getProject().getProjectProperties().getGlossaryRoot());
        priorityGlossary = new File(Core.getProject().getProjectProperties().getWriteableGlossary());
        monitor = new DirectoryMonitor(dir, this);
        monitor.start();
    }
```

Reply:
Initialization of priorotyGlossary is in line 139
```
        priorityGlossary = new File(Core.getProject().getProjectProperties().getWriteableGlossary());
```

It is good habit to move the check `boolean isPriority = priorityGlossary != null && priorityGlossary.equals(file);`  near the initialization, on the same screen,  then developers can easily understand why we need to check it.

Reply:
That makes sense, thank you very much for taking the time to explain